marxs/optics/multiLayerMirror.py

```diff
@@ -92,8 +94,11 @@ def process_photons(self, photons):
         spread_refl = np.interp(local_intersection[:,1], local_coords_in_file, reflectFile['FWHM(nm)'])
         
         wavelength = 1.23984282 / photons['energy']   # wavelength is in nm assuming energy is in keV
-        c_squared = (spread_refl ** 2) / (2 * np.log(2))
+        c_squared = (spread_refl ** 2) / (2. * np.log(2))
+        c_is_zero = (c_squared == 0)
+        c_squared += c_is_zero * self.precision
```

Comment:
I see what you are doing here, but I'm not sure that is the best way of doing it.
My understanding is this `spread_refl` can sometimes be 0 and if it is, then the line after this comment makes problems because of a divide by 0. Thus, you add a small number to `c_squared` to avoid the"divide by 0 error". I suggest a simpler way, which would also mean that you don't need the `precision` parameter. (Fewer parameters is always better, because there is less potential for confusion what those parameters actually do.)

why don't you just do
```
photons['probability'][c_is_zero] = 0

refl_prob[~c_is_zero]= max_refl[~c_is_zero] * np.exp(-((wavelength[~c_is_zero] - peak_wavelength[~c_is_zero]) ** 2) / (2 * c_squared[~c_is_zero]))
refl_prob[~c_is_zero] *= np.einsum('ij,ij->i', polarization[~c_is_zero], v_1[~c_is_zero])**2]

etc.
```